src/openforms/submissions/logic/actions.py

```diff
@@ -263,10 +266,33 @@ def get_action_log_data(
         }
 
 
-ACTION_TYPE_MAPPING: Mapping[str, Type[ActionOperation]] = {
+@dataclass
+class SetRegistrationBackendAction(ActionOperation):
+    registration_backend_key: str
+
+    @classmethod
+    def from_action(cls, action: ActionDict) -> "SetRegistrationBackendAction":
+        return cls(registration_backend_key=action["action"]["value"])
+
+    def apply(
+        self, step: SubmissionStep, configuration: FormioConfigurationWrapper
+    ) -> None:
+        step.submission.registration_backend = self.registration_backend_key
```

Comment:
`str` is a FormRegistrationBackendKey
since (form_id, backend_key) is a compound key, it's enough for form logic to do a cheap operation of setting such a "str"

On finalisation it's resolved to a FormRegistrationBackend.

I've changed the cryptic `str` to a NewType to express the intent. With algebraic data types, the match would be much cleaner, but on runtime it's still just a `str`.

Reply:
related to the delete and `models.SET_NULL`, I'm thinking it might be the best option to just use a CharField on the model and store the key, exactly because the combination of `(form_id, key)` uniquely identifies the backend from the model. Do keep in mind that it's possible the key no longer exists (so we don't get referential integrity with this approach).